modules/post/windows/manage/exec_powershell.rb

```diff
@@ -0,0 +1,61 @@
+
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'msf/core/post/windows/powershell'
+
+class Metasploit3 < Msf::Post
+  include Msf::Post::Windows::Powershell
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'                 => "Windows Powershell Execution Post Module",
+      'Description'          => %q{
+        This module will execute a powershell script in a meterpreter session.
+        The user may also enter text substitutions to be made in memory before execution.
+        Setting VERBOSE to true will output both the script prior to execution and the results.
+      },
+      'License'              => MSF_LICENSE,
+      'Platform'             => ['windows'],
+      'SessionTypes'         => ['meterpreter'],
+      'Author'               => [
+        'Nicholas Nam (nick[at]executionflow.org)', # original meterpreter script
+        'RageLtMan' # post module and libs
+        ]
+    ))
+
+    register_options(
+      [
+        OptString.new( 'SCRIPT',  [true, 'Path to the PS script or command string to execute' ]),
```

Comment:
My last merge should have updated all the EOF calls. It was actually in the
original design but we lost them in master along the way. The Rex end
should be able to take a nil. I've been slammed for the last month, but can
take another look to assess/address comments tomorrow evening.
On Aug 16, 2015 5:49 AM, "Ben Turner" <notifications@github.com> wrote:

> In modules/post/windows/manage/exec_powershell.rb
> <https://github.com/rapid7/metasploit-framework/pull/5379#discussion_r37144208>
> :
>
> > +        This module will execute a powershell script in a meterpreter session.
> > +        The user may also enter text substitutions to be made in memory before execution.
> > +        Setting VERBOSE to true will output both the script prior to execution and the results.
> > +      },
> > +      'License'              => MSF_LICENSE,
> > +      'Platform'             => ['windows'],
> > +      'SessionTypes'         => ['meterpreter'],
> > +      'Author'               => [
> > +        'Nicholas Nam (nick[at]executionflow.org)', # original meterpreter script
> > +        'RageLtMan' # post module and libs
> > +        ]
> > +    ))
> > +
> > +    register_options(
> > +      [
> > +        OptString.new( 'SCRIPT',  [true, 'Path to the PS script or command string to execute' ]),
>
> Changed the script description to include local, changes reflected in
> 8800d89
> <https://github.com/rapid7/metasploit-framework/commit/8800d894243ab2a7fd39bb398d3fff5bedd83367>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rapid7/metasploit-framework/pull/5379/files#r37144208>
> .
>